app/graphql/sources/person_current_roles_source.rb

```diff
@@ -0,0 +1,51 @@
+module Sources
+  class PersonCurrentRolesSource < GraphQL::Dataloader::Source
+    def fetch(person_content_ids)
+      all_roles = Edition
+        .live
+        .includes(
+          document: {
+            reverse_links: { # role -> role_appointment
+              link_set: {
+                documents: [
+                  :editions, # role_appointment
+                  :link_set_links, # role_appointment -> person
+                ],
+              },
+            },
+          },
+        )
+        .where(
+          document_type: "ministerial_role",
+          document: { locale: "en" },
+          reverse_links: { link_type: "role" },
+          documents: { locale: "en" }, # role_appointment Documents
+          editions_documents: { document_type: "role_appointment" }, # role_appointment Editions
+          link_set_links: { target_content_id: person_content_ids, link_type: "person" },
+        )
+        .where("editions_documents.details ->> 'current' = 'true'") # editions_documents is the alias that Active Record gives to the role_appointment Editions in the SQL query
+        .order(reverse_links: { position: :asc })
+
+      ids_map = person_content_ids.index_with { [] }
+
+      all_roles.each_with_object(ids_map) { |role, hash|
```

Comment:
I still don't have the best mental model for how this stuff is working and it's hurting my brain a little bit. I'm reasonably confident in it but i don't have time left this afternoon to get into my head well enough to approve. @brucebolt if you're more sure are you okay to give the 👍 ?

Commenting here because I'm reasonably confident this section is key as it's the bit that's flattening the layers out and doing the actual heavy lifting?

Reply:
I'm not quite 100% confident in my understanding of how this all fits together, but it seems to make sense from my perspective.

It's also covered by the ministers index integration spec.